src/controllers/UserController.js

```diff
@@ -26,4 +26,41 @@ export default class UserController {
       return res.status(500).json({ error: error.message });
     }
   }
+
+  /**
+ * @name getUser
+ * @description Allows user to sign in user
+ * @param {object} req The request object
+ * @param {object} res The response object
+ * @returns {object} The API response
+ */
+  static async getUser(req, res) {
+    try {
+      const { email, password } = req.body;
+      const user = await UserServices.loginUser(email);
+
+      if (user && Utils.comparePassword(password, user.password)) {
+        delete user.password;
+        return res.status(200).json({
+          status: 200,
+          data: user
+        });
+      }
+      if (user && !Utils.comparePassword(password, user.password)) {
+        return res.status(400).json({
+          status: 400,
+          error: 'Inavalid email/password.'
+        });
+      }
+    } catch (error) {
+      let err = '';
+      if (error.message === 'Illegal arguments: string, undefined') {
+        err = 'User does not exist.';
```

Comment:
I think you should also check for user existence in a middleware, there's already one for that in the code.

Reply:
I will suggest you use the user exist middleware for this, and add the middleware method to your sign in route....

Reply:
Please first check if a user is found in the db, if not, return an error immediately, else you can continue with the rest of the logic

Reply:
Done so